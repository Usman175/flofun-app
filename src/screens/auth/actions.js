import {
  createUser,
  requestOneTimePassword,
  verifyOneTimePassword,
} from '../../../constants/api'

export const CHANGE_PHONE = 'CHANGE_PHONE'
export const CHANGE_CODE = 'CHANGE_CODE'
export const START_CREATE_USER = 'START_CREATE_USER'
export const DONE_CREATE_USER = 'DONE_CREATE_USER'
export const DONE_VALIDATE_CODE = 'DONE_VALIDATE_CODE'

export const handleChangePhone = (phone) => ({
  type: CHANGE_PHONE,
  payload: phone,
})

export const handleChangeCode = (code) => ({
  type: CHANGE_CODE,
  payload: code,
})

export const startCreateUserAndSendCode = () => ({
  type: START_CREATE_USER,
})

export const doneCreateUserAndSendCode = (phone) => ({
  type: DONE_CREATE_USER,
  payload: phone,
})

export const createUserAndSendCode = (phone, navigation) => async (dispatch) => {
  dispatch(startCreateUserAndSendCode())

  try {
    const { success: createUserSuccess } = await createUser(`+1${phone}`)
    if (createUserSuccess) {
      const { success: requestOneTimePasswordSuccess } = await requestOneTimePassword(`+1${phone}`)
      if (requestOneTimePasswordSuccess) {
        dispatch(doneCreateUserAndSendCode(phone))
        navigation.navigate('EnterCodeScreen')
      } else {
        // TODO
      }
    } else {
      // TODO
    }
  } catch (err) {
    console.log(err) //eslint-disable-line
  }
}

export const doneValidateCode = (phone, token) => ({
  type: DONE_VALIDATE_CODE,
  payload: { phone, token },
})

export const validateCode = (phone, code) => async (dispatch) => {
  try {
    const { success, token } = await verifyOneTimePassword(`+1${phone}`, code)

    if (success) {
      dispatch(doneValidateCode(phone, token))
    }
  } catch (err) {
    console.log(err) //eslint-disable-line
  }
}
