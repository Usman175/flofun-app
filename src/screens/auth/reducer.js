export default (state = { logged: true }, action) => {
  switch (action.type) {
    default:
      return state
  }
}
