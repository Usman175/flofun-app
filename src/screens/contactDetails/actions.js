export const CHANGE_SENDER = 'CHANGE_SENDER'
export const CHANGE_RECIPIENT = 'CHANGE_RECIPIENT'

export const handleChangeSender = sender => ({
  type: CHANGE_SENDER,
  payload: sender,
})

export const handleChangeRecipient = recipient => ({
  type: CHANGE_RECIPIENT,
  payload: recipient,
})
