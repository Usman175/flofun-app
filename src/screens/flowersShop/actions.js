import { fetchFlowers } from '../../../constants/api'

export const FETCH_FLOWERS = 'FETCH_FLOWERS'
export const SELECT_FLOWER = 'SELECT_FLOWER'
export const REMOVE_FLOWER = 'REMOVE_FLOWER'
export const CHANGE_MESSAGE = 'CHANGE_MESSAGE'

export const fetchAvailableFlowers = () => ({
  type: FETCH_FLOWERS,
  payload: fetchFlowers(),
})

export const selectFlower = flower => ({
  type: SELECT_FLOWER,
  payload: flower,
})

export const removeSelectedFlower = flower => ({
  type: REMOVE_FLOWER,
  payload: flower,
})

export const handleChangeMessage = (message, key) => ({
  type: CHANGE_MESSAGE,
  payload: { message, key },
})
