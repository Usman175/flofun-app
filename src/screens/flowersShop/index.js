import FlowerShopScreen from './FlowerShopScreen'
import FlowerShopReducer from './reducer'

export {
  FlowerShopScreen,
  FlowerShopReducer,
}
