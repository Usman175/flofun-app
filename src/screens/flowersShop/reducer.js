import {
  FETCH_FLOWERS,
  SELECT_FLOWER,
  REMOVE_FLOWER,
  CHANGE_MESSAGE,
} from './actions'

const INITIAL_STATE = {
  selectedFlowers: [],
  flowers: {
    data: [],
    isFetched: false,
    error: {
      on: false,
      message: null,
    },
  },
}

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case `${FETCH_FLOWERS}_FULFILLED`:
      return {
        ...state,
        flowers: {
          data: action.payload.map((f) => Object.assign(f, { quantity: 1 })),
          isFetched: true,
          error: {
            on: false,
            message: null,
          },
        },
      }
    case `${FETCH_FLOWERS}_REJECT`:
      return {
        ...state,
        flowers: {
          data: [],
          isFetched: true,
          error: {
            on: true,
            message: 'Error when fetching flowers',
          },
        },
      }
    case SELECT_FLOWER: {
      const tmpSelectedFlowers = state.selectedFlowers.slice()
      let modalMessage = ''
      if (tmpSelectedFlowers.length === 3) {
        modalMessage = 'Cart Full'
      } else {
        tmpSelectedFlowers.push({ ...action.payload, message: '' })
      }

      return {
        ...state,
        selectedFlowers: tmpSelectedFlowers,
        modalMessage,
      }
    }
    case REMOVE_FLOWER: {
      const newSelectedFlowers = state.selectedFlowers.filter((f) => f._id !== action.payload._id)

      return {
        ...state,
        selectedFlowers: newSelectedFlowers,
      }
    }
    case CHANGE_MESSAGE: {
      const tmpSelectedFlowers = state.selectedFlowers.slice()
      tmpSelectedFlowers[action.payload.key].message = action.payload.message

      return {
        ...state,
        selectedFlowers: tmpSelectedFlowers,
      }
    }
    default:
      return state
  }
}
