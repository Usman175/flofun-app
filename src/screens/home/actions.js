import { fetchFlowers } from '../../../constants/api'

export const FETCH_FLOWERS = 'FETCH_FLOWERS'

export const fetchAvailableFlowers = () => ({
  type: FETCH_FLOWERS,
  payload: fetchFlowers(),
})
