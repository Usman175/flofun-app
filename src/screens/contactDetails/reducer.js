import { CHANGE_SENDER, CHANGE_RECIPIENT } from './actions'

const INITIAL_STATE = {
  senderName: '',
  recipientName: '',
}

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case CHANGE_SENDER:
      return {
        ...state,
        senderName: action.payload,
      }
    case CHANGE_RECIPIENT:
      return {
        ...state,
        recipientName: action.payload,
      }
    default:
      return state
  }
}
