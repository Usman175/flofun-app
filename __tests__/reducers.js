import { FETCH_FLOWERS } from '../src/screens/flowersShop/actions'
import reducer from '../src/redux/reducers'

const flowers = [
  { title: 'flower 1', image: 'img1', price: 30 },
  { title: 'flower 2', image: 'img2', price: 40 },
  { title: 'flower 3', image: 'img3', price: 50 },
]

describe('FlowersShop Reducer', () => {
  it('should return the initial state', () => {
    expect(
      reducer(undefined, {}).flowerShop
    ).toMatchSnapshot()
  })

  it('should return the initial if pending', () => {
    expect(
      reducer(undefined, {
        type: `${FETCH_FLOWERS}_PENDING`,
      }).flowerShop
    ).toMatchSnapshot()
  })

  it('should fill the data if fullfiled', () => {
    expect(
      reducer(undefined, {
        type: `${FETCH_FLOWERS}_FULFILLED`,
        payload: flowers,
      }).flowerShop
    ).toMatchSnapshot()
  })

  it('should return the error state if rejected', () => {
    expect(
      reducer(undefined, {
        type: `${FETCH_FLOWERS}_REJECT`,
      }).flowerShop
    ).toMatchSnapshot()
  })
})
