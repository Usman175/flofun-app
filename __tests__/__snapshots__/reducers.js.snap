// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Reducer should fill the data if fullfiled 1`] = `
Object {
  "flowers": Object {
    "data": Array [
      Object {
        "image": "img1",
        "price": 30,
        "title": "flower 1",
      },
      Object {
        "image": "img2",
        "price": 40,
        "title": "flower 2",
      },
      Object {
        "image": "img3",
        "price": 50,
        "title": "flower 3",
      },
    ],
    "error": Object {
      "message": null,
      "on": false,
    },
    "isFetched": true,
  },
}
`;

exports[`Home Reducer should return the error state if rejected 1`] = `
Object {
  "flowers": Object {
    "data": Array [],
    "error": Object {
      "message": "Error when fetching flowers",
      "on": true,
    },
    "isFetched": true,
  },
}
`;

exports[`Home Reducer should return the initial if pending 1`] = `
Object {
  "flowers": Object {
    "data": Array [],
    "error": Object {
      "message": null,
      "on": false,
    },
    "isFetched": false,
  },
}
`;

exports[`Home Reducer should return the initial state 1`] = `
Object {
  "flowers": Object {
    "data": Array [],
    "error": Object {
      "message": null,
      "on": false,
    },
    "isFetched": false,
  },
}
`;
